src/function/arithmetic/round.js

```diff
@@ -67,9 +71,21 @@ export const createRound = /* #__PURE__ */ factory(name, dependencies, ({ typed,
    * @return {number | BigNumber | Fraction | Complex | Array | Matrix} Rounded value
    */
   return typed(name, {
-    number: roundNumber,
+    number: function (x) {
+      if (nearlyEqual(x, roundNumber(x, epsilonExponent), config.epsilon)) {
+        return roundNumber(roundNumber(x, epsilonExponent))
+      } else {
+        return roundNumber
```

Comment:
I think this should be `return roundNumber(x)`? If so, we should add a unit test that covers this branch.